cmd/dlt/externalauthproviders/cmd.go

```diff
@@ -0,0 +1,55 @@
+/*
+Copyright (c) 2024 Red Hat, Inc.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+  http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package externalauthproviders
+
+import (
+	"github.com/openshift/rosa/pkg/interactive/confirm"
+	"github.com/openshift/rosa/pkg/ocm"
+	"github.com/openshift/rosa/pkg/rosa"
+	"github.com/spf13/cobra"
+)
+
+var Cmd = &cobra.Command{
+	Use:     "external-auth-providers",
+	Aliases: []string{"externalauthproviders", "externalauthprovider", "external-auth-provider"},
+	Short:   "Delete External Authentication Providers",
+	Long:    "Cleans up external authentication providers for the selected cluster.",
+	Example: `  # Delete External Authentication Providers"
+  rosa delete external-auth-providers with TBD param`,
+	Run: run,
+}
+
+func init() {
+	ocm.AddClusterFlag(Cmd)
+	confirm.AddFlag(Cmd.Flags())
+}
+
+func run(_ *cobra.Command, _ []string) {
+	r := rosa.NewRuntime().WithOCM()
+	defer r.Cleanup()
+
+	clusterKey := r.GetClusterKey()
+	cluster := r.FetchCluster()
+
+	r.Reporter.Debugf("Deleting External Authentication Providers for cluster '%s'", clusterKey)
+
+	if cluster.Hypershift().Enabled() {
+		r.OCMClient.DeleteExternalAuthProviders(clusterKey, "test")
+	} else {
+		r.Reporter.Errorf("External authentication provider is only supported for Hosted Control Planes.")
+	}
```

Comment:
Suggest to move this to a func that returns an error IMO, so it can be easier to test.